src/nbl/video/COpenGLConnection.h

```diff
@@ -0,0 +1,48 @@
+#ifndef __NBL_C_OPENGL_CONNECTION_H_INCLUDED__
+#define __NBL_C_OPENGL_CONNECTION_H_INCLUDED__
+
+#include "nbl/video/IAPIConnection.h"
+#include "nbl/video/CEGL.h"
+#include "nbl/video/COpenGLPhysicalDevice.h"
+
+namespace nbl {
+namespace video
+{
+
+class COpenGLConnection final : public IAPIConnection
+{
+public:
+    COpenGLConnection()
+    {
+        m_egl.initialize();
+        m_pdevice = COpenGLPhysicalDevice::create(&m_egl);
+    }
```

Comment:
I recommend core::smart_refctd_ptr<CVulkanConnection> create static method because if anything goes wrong in the constructor you're fucked (you have to throw an exception)

Reply:
yea, theres a design for such